nixos/modules/services/mail/mailpit.nix

```diff
@@ -0,0 +1,103 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.mailpit;
+  inherit (lib)
+    cli
+    concatStringsSep
+    const
+    filterAttrs
+    getExe
+    mkEnableOption
+    mkIf
+    mkOption
+    types
+    ;
+
+  isNonNull = v: v != null;
+  cliFlags = concatStringsSep " " (
+    cli.toGNUCommandLine { } (filterAttrs (const isNonNull) cfg.settings)
+  );
+in
+{
+  options.services.mailpit = {
+    enable = mkEnableOption "mailpit";
+
+    settings = mkOption {
+      default = { };
+      type = types.submodule {
+        freeformType = types.attrsOf (
+          types.oneOf [
+            types.str
+            types.int
+            types.bool
+          ]
+        );
+        options = {
+          db-file = mkOption {
```

Comment:
This seems to be outdated, but as nixpkgs has a quite old version of mailpit it seems correct for now..

Reply:
Oof I already wondered why how I came up with `database` while hacking on that module 🤦 
I guess I'll do an update of the software then, thanks for catching this.

Reply:
FYI I already opened https://github.com/NixOS/nixpkgs/pull/324302 for automatic update PRs via @r-ryantm 
Still waiting to get it merged.

Btw I also initiated that the mailpit documentation shows the version in which specific options were added. See https://github.com/axllent/mailpit/issues/328 for more info.